src/core/server/plugins/plugins_service.ts

```diff
@@ -107,6 +121,27 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
     await this.pluginsSystem.stopPlugins();
   }
 
+  private generateUiPluginsConfigs(
+    uiPlugins: Map<string, DiscoveredPlugin>
+  ): Map<PluginName, Observable<unknown>> {
+    return new Map(
+      [...uiPlugins].map(([pluginId, plugin]) => {
+        const configDescriptor = this.pluginConfigDescriptors.get(pluginId);
+        if (configDescriptor && configDescriptor.exposeToBrowser) {
+          return [
+            pluginId,
+            this.configService
+              .atPath(plugin.configPath)
+              .pipe(
+                map((config: any) => pick(config || {}, configDescriptor.exposeToBrowser || []))
```

Comment:
nit: `exposeToBrowser` cannot be falsy. https://github.com/elastic/kibana/pull/50641/commits/daf09e3be50a80716baf2698d8a2275b4f8c792b#diff-725c498a267472754f01cff2baa3901bR130